packages/ui-top-nav-bar/src/TopNavBar/utils/mapItemsForDrilldown.tsx

```diff
@@ -62,7 +59,7 @@ const mapItemsForDrilldown = (
   const submenus: ItemMappedForDrilldownOption[] = []
   const { currentPageId, renderOptionContent } = options
 
-  const customPopoverIdMap = generateInstanceCounterMap()
+  const customPopoverIdMap = new Map<string, number>()
```

Comment:
using the global counter caused here a weird bug, I think because it was re-rendered multiple times the IDs got out of sync causing the menu buttons in SmallViewport mode to break.